src/smk/viewer-leaflet/layer/layer-vector-leaflet.js

```diff
@@ -154,14 +154,36 @@ include.module( 'layer-leaflet.layer-vector-leaflet-js', [ 'layer.layer-vector-j
                                 layers[ 0 ].changedFeature( { newPt: { latitude: ll.lat, longitude: ll.lng }, geojson: geojson } )
                             } )
                     },
-                    onEachFeature: function ( feature, layer ) {
-                        if ( layer.setStyle )
-                            layer.setStyle( convertStyle( feature.style /* || layers[ 0 ].config.style */, feature.geometry.type ) )
-                    },
                     renderer: L.svg(),
                     interactive: false,
                     pane: layerPane
-                } )
+                };
+                
+                if (layers[0].config.conditionalStyles) {
+                    const defaultStyle = layers[0].config.style ? layers[0].config.style : {};
+                    layerOptions.style = function(feature) {
+                        var combinedStyle = Object.assign({}, defaultStyle);
+                        layers[0].config.conditionalStyles.forEach(conditionalStyle => {
+                            if (!Object.keys(feature.properties).includes(conditionalStyle.property)) {
+                                console.debug(`The feature property ${conditionalStyle.property} was not found; conditional styling will not be applied for this property.`);
+                                return convertStyle(combinedStyle, feature.geometry.type);
+                            }
+                            conditionalStyle.conditions.forEach(condition => {
+                                if (condition.value === feature.properties[conditionalStyle.property]) {
+                                    Object.assign(combinedStyle, condition.style);
+                                    return;
```

Comment:
Overall I think this is a very reasonable approach. You're covering a very common use case. The 'Esriesque' term for this would be a unique value renderer. The esri-leaflet package may already have similar functionality out of the box for Esri feature layers.

I could be understanding things incorrectly, but there might be a little bit excess work being done in lines 171-174. What is the purpose of the return statement on line 174? On line 171 you use 'forEach' to iterate through all conditions, but I think you can actually stop once you find a matching condition. If this is the case, you can't 'break out of' a forEach 'loop' as it is a function, not a loop. You could replace the forEach call with a for loop which you can break out of once you find a matching condition, or filter your array of conditions first.

Reply:
Thanks!

I like your suggestion of filtering conditions and will push a change implementing it.